src/Illuminate/Support/Str.php

```diff
@@ -403,7 +403,7 @@ public static function snake($value, $delimiter = '_')
     public static function startsWith($haystack, $needles)
     {
         foreach ((array) $needles as $needle) {
-            if ($needle != '' && mb_strpos($haystack, $needle) === 0) {
+            if ($needle != '' && substr($haystack, 0, strlen($needle)) === $needle) {
```

Comment:
@vlakoff strpos runs from the start to the end of the string regardless of where the match (or non-match) exists, if you have a large string (i.e. 4.5MB as in my example) you have to check the entire string which is inefficient since we only care about the first strlen($needle) bytes.